### PR TITLE
Ensure that the condor user is the same across containers

### PR DESCRIPTION
**Pull request by brianhlin**
Using the RedHat assigned UID/GID for the condor user

### CODE REVIEW

File: build/docker/services/base/Dockerfile [hunk 1]
```diff
@@ -15,6 +15,11 @@ LABEL org.label-schema.name="htcondor/base:${VERSION}-el${EL}${SUFFIX}" \
       org.label-schema.vendor="HTCondor" \
       org.label-schema.license="Apache-2.0"
 
+# Ensure that the 'condor' UID/GID matches across containers
+RUN groupadd -g 64 -r condor
+RUN useradd -r -g condor -d /var/lib/condor -s /sbin/nologin \
+    -u 64 -c "Owner of HTCondor Daemons" condor
```
Comment (matyasselmeci) [inline build/docker/services/base/Dockerfile:21]: No point in separate layers for the groupadd and the useradd.
```suggestion
RUN groupadd -g 64 -r condor && \
    useradd -r -g condor -d /var/lib/condor -s /sbin/nologin \
      -u 64 -c "Owner of HTCondor Daemons" condor
```

**Review by matyasselmeci — APPROVED** [review]
Minor change but otherwise fine.

**Review by timtheisen — APPROVED** [review]
Combine the groupadd and useradd into a single RUN.

**Comment by matyasselmeci** [comment]
We'll need to update https://github.com/htcondor/htcondor/blob/master/build/docker/k8s/pool.yaml#L120-L123 after this gets merged.

**Comment by mvpel** [comment]
I think it would be a better approach to refer to the uidgid file in the setup package on RedHat/CentOS rather than hard coding 64, and default to 64 on other distros where that file is not found. 

https://access.redhat.com/documentation/en-us/red_hat_enterprise_linux/6/html/migration_planning_guide/sect-migration_guide-security_authentication-system_users

**Comment by matyasselmeci** [comment]
We need to have a hardcoded value in order to be able to refer to that user by UID. For example, the yaml file I linked to involves running a pod with an htcondor/exec container and a docker-in-docker container as a sidecar (for running docker universe jobs). The docker-in-docker container needs to use the uid/gid of the condor user but they don't share a passwd file so it can't do the mapping itself.

**Comment by brianhlin** [comment]
I think @mvpel is suggesting that we rely on RedHat and lookup the static UID/GID at build-time. One downside I can see to that is if the values ever change upstream, then we can lose interoperability with older images and muck up permissions for users' volume mounts.

**Comment by matyasselmeci** [comment]
Also it would be nice if it were easy to determine the UID/GID by inspection of the Dockerfile, instead of having to visit a website.

**Comment by brianhlin** [comment]
@matyasselmeci @timtheisen what else needs to be done before merging this? What triggers new container builds, i.e. when can I start pulling containers that'll contain these changes?

**Comment by timtheisen** [comment]
I'll merge this presently.

**Comment by matyasselmeci** [comment]
Don't forget to update https://github.com/htcondor/htcondor/blob/master/build/docker/k8s/pool.yaml#L120-L123.

**Pull request merged**